projects/plugins/jetpack/extensions/blocks/payment-buttons/edit.js

```diff
@@ -65,18 +65,23 @@ function PaymentButtonsEdit( { clientId } ) {
 	const showStripeConnectAction = ! shouldUpgrade && ! isApiConnected && !! connectUrl;
 
 	const blockProps = useBlockProps();
-	const innerBlocksProps = useInnerBlocksProps(
-		{},
-		{
-			allowedBlocks: ALLOWED_BLOCKS,
-			orientation: 'horizontal',
-			template: [ [ 'jetpack/recurring-payments' ] ],
-			templateInsertUpdatesSelection: true,
-		}
-	);
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		allowedBlocks: ALLOWED_BLOCKS,
+		orientation: 'horizontal',
+		template: [ [ 'jetpack/recurring-payments' ] ],
+		templateInsertUpdatesSelection: true,
+		__experimentalLayout: layout,
+	} );
 
+	// The ID needs to be just on the outermost wrapper - the toolbar and wpcom upgrade nudge
+	// will then be positioned in relation to this.
+	delete innerBlocksProps.id;
```

Comment:
I think it still works without deleting it from the inner block, but not deleting it means that there are two elements (inner and outer) with the same HTML id, which violates the HTML spec. Also it loses symmetry with data-block (which is a prefixed version of the id)

Reply:
Oh right, I see it now. It's a side effect of using separately `blockProps` and `innerBlocksProps`, since `innerBlocksProps` extends from `blockProps`.